Corrfunc/tests/test_theory.py

```diff
@@ -178,6 +178,47 @@ def test_duplicate_cellpairs(autocorr, binref, min_sep_opt, maxcells,
     assert np.all(results['npairs'] == [0, 2])
 
 
+@pytest.mark.parametrize('autocorr', [0, 1])
+@pytest.mark.parametrize('binref', [1, 2, 3])
+@pytest.mark.parametrize('min_sep_opt', [False, True])
+@pytest.mark.parametrize('maxcells', [1, 2, 3])
+def test_rmax_against_brute(autocorr, binref, min_sep_opt, maxcells,
+                            isa='fastest', nthreads=maxthreads()):
+    '''Generate 50 particles near (0,0,0) and 50 near (0,0,L/2)
+    and compare against the brute-force answer.
+
+    Use close to the max allowable Rmax, 0.49*Lbox.
+    '''
+    np.random.seed(1234)
+    pos = np.empty((3, 100), dtype='f8')
+    eps = 0.2
+    boxsize = 123.
+    bins = np.linspace(0.01, 0.49*boxsize, 20)
+
+    pos[:, :50] = np.random.uniform(low=-eps, high=eps, size=(3, 50))*boxsize
+    pos[:, 50:] = np.random.uniform(low=-eps, high=eps,
+                                    size=(3, 50))*boxsize + boxsize/2.
+
+    pdiff = np.abs(pos[:, None] - pos[:, :, None])
+    pdiff[pdiff >= boxsize/2] -= boxsize
+    pdiff = (pdiff**2).sum(axis=0).reshape(-1)
+    brutecounts, _ = np.histogram(pdiff, bins=bins**2)
+    assert brutecounts[0] == 2
+    assert brutecounts[-1] == 172
```

Comment:
This is a bit too hard-coded - could use `npts` as a variable. I also don't follow the first set of positions - wouldn't line 198 generate positions that are between `[-eps, +eps]*boxsize` (rather than the more stringent [0, eps] and [boxsize-eps, boxsize])?

How did you get the rhs values in lines 206-207?

Reply:
I should have left a comment; that check was strictly to verify that the comparison with Corrfunc was actually doing something, and not trivially passing due to returning all zeros.  The real test is the comparison with Corrfunc which follows below.

Reply:
On the positions, I tried to generate a cloud around (0,0,0) and (L/2,L/2,L/2). Each cloud has width `0.2*boxsize`.  The relatively large width is so that we can test pairs between the clouds---the two clouds must be within L/2 of each other on the diagonal!

Reply:
> I should have left a comment; that check was strictly to verify that the comparison with Corrfunc was actually doing something, and not trivially passing due to returning all zeros. The real test is the comparison with Corrfunc which follows below.

In that case checking with `assert  brutecounts[0] > 0 and brutecounts[-1] > 0` might be sufficient. Particularly, so if the specific counts might change based on the random number seed

Reply:
```python
import numpy as np
seed = 1234
np.random.seed(seed)

eps = 0.2
boxsize = 123.
npts = 10
pos = np.empty((3, npts), dtype='f8')
pos[:, :npts//2] = np.random.uniform(low=-eps, high=eps, size=(3, npts//2))*boxsize
pos[:, npts//2:] = np.random.uniform(low=-eps, high=eps, size=(3, npts//2))*boxsize + boxsize/2.

pos

array([[-1.51772430e+01,  6.00775154e+00, -3.06379524e+00,
         1.40396423e+01,  1.37748098e+01,  6.45108524e+01,
         6.16516917e+01,  3.75774077e+01,  7.49230698e+01,
         8.03259466e+01],
       [-1.11884438e+01, -1.09979586e+01,  1.48521111e+01,
         2.25404562e+01,  1.84958856e+01,  5.48523904e+01,
         6.71774920e+01,  4.06087571e+01,  5.50461411e+01,
         8.28104930e+01],
       [-6.99539032e+00,  4.89601758e-02,  9.02637641e+00,
         1.04649397e+01, -6.38366286e+00,  6.89478046e+01,
         5.64423668e+01,  7.57055230e+01,  5.24883372e+01,
         6.48504537e+01]])
```
That snippet produces invalid positions - i.e., positions not in `[0, Lbox)`. Plus, the width of the distribution (which should be ~ `eps`) is multiplied by the boxsize. We can get the expected behaviour by removing the multiplication with boxsize (or change the semantics of `eps` such that `eps` is a fraction of boxsize -> generate the random positions from `-eps*boxsize, +eps*boxsize`) and then also wrapping the positions into `[0, Lbox)`.

Here's my attempt:

```python
import numpy as np
seed = 1234
np.random.seed(seed)

# eps is the width of the point clouds as a fraction of boxsize
eps = 0.01 
boxsize = 123.
npts = 10
pos = np.empty((3, npts), dtype='f8')
pos[:, :npts//2] = np.random.uniform(low=-eps*boxsize, high=eps*boxsize, size=(3, npts//2))
pos[:, npts//2:] = np.random.uniform(low=-eps*boxsize, high=eps*boxsize, size=(3, npts//2)) + boxsize/2.

pos[pos < 0] += boxsize
pos[pos >= boxsize] -= boxsize

pos

array([[1.22241138e+02, 3.00387577e-01, 1.22846810e+02, 7.01982116e-01,
        6.88740488e-01, 6.16505426e+01, 6.15075846e+01, 6.03038704e+01,
        6.21711535e+01, 6.24412973e+01],
       [1.22440578e+02, 1.22450102e+02, 7.42605557e-01, 1.12702281e+00,
        9.24794281e-01, 6.11676195e+01, 6.17838746e+01, 6.04554379e+01,
        6.11773071e+01, 6.25655247e+01],
       [1.22650230e+02, 2.44800879e-03, 4.51318821e-01, 5.23246986e-01,
        1.22680817e+02, 6.18723902e+01, 6.12471183e+01, 6.22102762e+01,
        6.10494169e+01, 6.16675227e+01]])
```

What do you think?